cmd/controller/main.go

```diff
@@ -43,6 +54,11 @@ func main() {
 	lo.Must0(op.AddHealthzCheck("cloud-provider", awsCloudProvider.LivenessProbe))
 	cloudProvider := metrics.Decorate(awsCloudProvider)
 
+	err := validateK8sVersion(ctx, op)
```

Comment:
musing: Looking at the arguments that this function takes in makes me naturally ask the question if this should be a method on the operator instead of take the operator in as an argument. e.g. `op.ValidateK8sVersion(ctx)`

Reply:
See: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/version#MustParseGeneric